src/azure-cli/azure/cli/command_modules/servicefabric/tests/latest/test_sf_cluster.py

```diff
@@ -60,6 +60,71 @@ def test_update_settings_and_reliability(self):
         self.cmd('sf cluster reliability update --resource-group {rg} -c {cluster_name} --reliability-level Silver',
                  checks=[self.check('reliabilityLevel', 'Silver')])
 
+    @ResourceGroupPreparer()
+    def test_cluster_settings_and_upgrade_mode(self):
+        self.kwargs.update({
+            'kv_name': self.create_random_name('sfrp-cli-kv-', 24),
+            'loc': 'westus',
+            'cert_name': self.create_random_name('sfrp-cli-', 24),
+            'cluster_name': self.create_random_name('sfrp-cli-', 24),
+            'vm_password': "Pass123!@#",
+            'primary_node_type': 'nt1vm',
+            'new_node_type': 'nt2',
+            'cluster_size': '5',
+            'version': '10.1.2338.9590'
+        })
+        _create_cluster(self, self.kwargs)
+        _wait_for_cluster_state_ready(self, self.kwargs)
+
+        # add setting
+        self.cmd('sf cluster update --resource-group {rg} -c {cluster_name} --settings-section-set --section NamingService --parameter MaxOperationTimeout --value 10001',
+                 checks=[self.check('length(fabricSettings)', 2),
+                         self.check('fabricSettings[1].name', 'NamingService'),
+                         self.check('fabricSettings[1].parameters[0].name', 'MaxOperationTimeout'),
+                         self.check('fabricSettings[1].parameters[0].value', '10001')])
+
+        # remove setting
+        self.cmd('sf cluster update --resource-group {rg} -c {cluster_name} --settings-section-rem --section NamingService --parameter MaxOperationTimeout',
+                 checks=[self.check('length(fabricSettings)', 1),
+                         self.check('fabricSettings[0].name', 'Security')])
+        
+        # update upgrade mode
+        self.cmd('sf cluster update --resource-group {rg} -c {cluster_name} --upgrade-mode manual --version {version}',
+                 checks=[self.check('upgradeMode', 'Manual')])
+        
+    @live_only()    
```

Comment:
May I ask why do you need to mark this test as `@live_only()`?

Reply:
after running test live it fails in playback with following errors and workaround to replaced mismatched values does not work, in this case keyvault name

E           AssertionError: Can't overwrite existing cassette ('e:\\repos\\azure-cli\\src\\azure-cli\\azure\\cli\\command_modules\\servicefabric\\tests\\latest\\recordings\\test_reliability_and_durability.yaml') in your current record mode ('once').
E           No match for the request (<Request (GET) https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.KeyVault/vaults/clitestrg000001?api-version=2023-02-01>) was found.
E           Found 3 similar requests with 1 different matcher(s) :
E
E           1 - (<Request (GET) https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.KeyVault/vaults/clitestrgojwj2hjwlnme?api-version=2023-02-01>).
E           Matchers succeeded : ['method', 'scheme', 'host', 'port', '_custom_request_query_matcher']
E           Matchers failed :
E           path - assertion failure :
E           /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.KeyVault/vaults/clitestrg000001 != /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.KeyVault/vaults/clitestrgojwj2hjwlnme
E
E           2 - (<Request (GET) https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.KeyVault/vaults/clitestrgojwj2hjwlnme?api-version=2023-02-01>).
E           Matchers succeeded : ['method', 'scheme', 'host', 'port', '_custom_request_query_matcher']
E           Matchers failed :
E           path - assertion failure :
E           /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.KeyVault/vaults/clitestrg000001 != /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.KeyVault/vaults/clitestrgojwj2hjwlnme
E
E           3 - (<Request (GET) https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.KeyVault/vaults/clitestrgojwj2hjwlnme?api-version=2023-02-01>).
E           Matchers succeeded : ['method', 'scheme', 'host', 'port', '_custom_request_query_matcher']
E           Matchers failed :
E           path - assertion failure :
E           /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.KeyVault/vaults/clitestrg000001 != /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.KeyVault/vaults/clitestrgojwj2hjwlnme

workaround of switching